### PR TITLE
Rarity badges instead of rarity stars

### LINKED ISSUE #127: Wrong poketypes shown on bottom sheet
After clicking a pokemon I can see its types correctly on the bottom sheet. However, after clicking another pokemon, these types won't change. It always shows the first pokemons types:

First click: Pidgey - `Normal` `Flying` (ok)
image

Second click: Ratata - `Normal` `Flying` (?)
image

Third click: Goldeen - `Normal` `Flying` (??)
image

### PR DESCRIPTION
**Pull request by mfkaptan**
- Use rectangular badges
- Red, orange and green for rare, uncommon and common respectively
- Update `poke-type` to take a single element instead of an array
- Fix type badge issues
- Fix #127 

issue #69 
can be used for #121

### CODE REVIEW

**Review by WoH — APPROVED** [review]

**Comment by mfkaptan** [comment]
Thanks for the review @WoH - the night coder!

**Comment by MajorBreakfast** [comment]
I think you should use a simple div instead of an `<ion-badge>`. The whole point of `<ion-badge>` is to get an appearance native to the operating system. That's about the opposite what we're using it for.

**Comment by mfkaptan** [comment]
How it looks like with the new changes:

image

image

image

File: ionic2/app/components/rarity-badge/rarity-badge.component.scss [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 }
 
-.uncommon {
-  background-color: orange;
-}
-
-.common {
-  background-color: green;
+div {
+  border-radius: 0px;
```
Comment (MajorBreakfast) [inline ionic2/app/components/rarity-badge/rarity-badge.component.scss]: Why?

File: ionic2/app/components/types/types.component.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
-<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typeArray">
+<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typesArray">
   {{ poketype.element }}
 </ion-badge>
```
Comment (MajorBreakfast) [inline ionic2/app/components/types/types.component.html]: I think you should make that a plain div as well

File: ionic2/app/components/types/types.component.html [hunk 3]
```diff
@@ -1,3 +1,3 @@
-<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typeArray">
+<ion-badge [style.background-color]="poketype.color" *ngFor="let poketype of typesArray">
```
Comment (MajorBreakfast) [inline ionic2/app/components/types/types.component.html]: Don't do the ngFor in the component, do it on the page

File: ionic2/app/components/types/types.component.scss [hunk 4]
```diff
@@ -1,4 +1,5 @@
 ion-badge{
```
Comment (MajorBreakfast) [inline ionic2/app/components/types/types.component.scss]: missing space

File: ionic2/app/components/types/types.component.scss [hunk 5]
```diff
@@ -1,4 +1,5 @@
...
   margin-left: 2.5px;
   margin-right: 2.5px;
+  margin-bottom: 2.5px;
```
Comment (MajorBreakfast) [inline ionic2/app/components/types/types.component.scss]: margin combo property might be better

File: ionic2/app/components/types/types.component.ts [hunk 6]
```diff
@@ -9,17 +9,17 @@ import { ApiService } from '../../services/api.service';
 })
 export class TypesComponent implements OnInit {
 
-  @Input() type: string[];
-  typeArray: Object[];
+  @Input() types: string[];
+  typesArray: Object[];
```
Comment (MajorBreakfast) [inline ionic2/app/components/types/types.component.ts]: In JavaScript it's uncustomary to end arrays with "-Array". Just use the plural

File: ionic2/app/components/types/types.component.ts [hunk 7]
```diff
@@ -9,17 +9,17 @@ import { ApiService } from '../../services/api.service';
...
 
   ngOnInit() {
     let typeColors = this.apiService.getTypes();
 
-    this.typeArray = [];
-    for (let item of this.type){
-      this.typeArray.push({element:item, color:typeColors[item]})
+    this.typesArray = [];
+    for (let item of this.types) {
+      this.typesArray.push({element:item, color:typeColors[item]})
```
Comment (MajorBreakfast) [inline ionic2/app/components/types/types.component.ts]: Don't modify it, make a new property

Comment (MajorBreakfast) [reply ionic2/app/components/types/types.component.ts]: You should take a look at `array.map(function (value) { return .... })`

File: ionic2/app/pages/poke-detail/poke-detail.page.html [hunk 8]
```diff
@@ -21,11 +25,8 @@
             <span *ngIf="pokemon.gender.abbreviation=='f'">, female</span>
             <span *ngIf="pokemon.gender.abbreviation=='m'">, male</span>
           </div>
-          <div id="types"><poke-type [type]="pokemon.types"></poke-type></div>
-          <!--div id="rarity">
-            <poke-rarity-badge [rarity]="pokemon.appearanceLikelihood"></poke-rarity-badge>
-            <span style="opacity: 0.5">(Rarity: Soon badge instead of stars)</span>
-          </div-->
+          <div id="types"><poke-type [types]="pokemon.types"></poke-type></div>
```
Comment (MajorBreakfast) [inline ionic2/app/pages/poke-detail/poke-detail.page.html]: Still bad. The component should be called `<poke-type>` it should have a `type` property and it should accept a single value (not an array)

File: ionic2/app/components/types/types.component.html (hunk 3 above)
Comment (mfkaptan) [reply ionic2/app/components/types/types.component.html]: May I ask what is the reason behind this? We have to change this in different places too. Couldn't it just stay like this?

File: ionic2/app/pages/poke-detail/poke-detail.page.html (hunk 8 above)
Comment (mfkaptan) [reply ionic2/app/pages/poke-detail/poke-detail.page.html]: Can we just change it to poke-types?

**Comment by MajorBreakfast** [comment]
The source code looks good. Feel feel to merge if everything works. Good work!

**Pull request merged**